google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -148,7 +150,7 @@ public Project getProject() {
           return Optional.empty();
         }
         if (runtimeOptional.filter(runtime -> runtime == FlexibleRuntime.CUSTOM).isPresent()
-            && !Files.exists(Paths.get(deploymentConfiguration.getDockerFilePath()))) {
+            && (!new File(deploymentConfiguration.getDockerDirectoryPath(), DOCKERFILE).exists())) {
```

Comment:
instead of exists() i'd suggest to use isFile() to make sure its not a directory.

But better yet, I think you can use the Paths api:

```
Path path = Paths.get(deploymentConfiguration.getDockerDirectoryPath(), DOCKERFILE));
..
.. && (!Files.isRegularFile(path))...
```

Same comment for all other cases doing this logic